### DISCUSSION TITLE
看到了大佬的Channel的视频，没理解他和队列的本质区别在哪里啊

### DISCUSSION THREAD

**Discussion by oufu99** [BYJRK/DotNet-Discussions#183] [C# 技术交流]
感觉视频里面讲的用队列都能做，而且也都能解耦，为啥不直接用队列?
我知道channel内部使用的concurrentQueue,但是他为啥要多封装一层?没理解这点。。。大佬能说一下不

**Comment by BYJRK** [comment]
`ConcurrentQueue` 的方法都是阻塞的，不适用于异步编程。而且你如果希望实现更多更复杂的特性，比如一或多个 `Reader`、`Writer`，或最大容量等，使用 `Channel` 都会明显更加容易配置一些。
